File: app/src/main/java/com/innso/mobile/ui/bindings/GeneralBindings.java
package com.innso.mobile.ui.bindings;

import android.content.Context;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.innso.mobile.InnsoApplication;
import com.innso.mobile.R;

import java.io.File;

import androidx.databinding.BindingAdapter;

public class GeneralBindings {

    @BindingAdapter("load_image")
    public static void loadImage(ImageView imageView, String imagePath) {
        Context context = InnsoApplication.get().getApplicationContext();
        Glide.with(context)
                .load(imagePath)
                .asBitmap()
                .placeholder(R.mipmap.ic_launcher)
                .error(R.mipmap.ic_launcher)
                .into(imageView);
    }

    @BindingAdapter("load_image_file")
    public static void loadImageFile(ImageView imageView, File fileImage) {
        if (fileImage != null) {
            Glide.with(imageView.getContext()).load(fileImage).into(imageView);
        }
    }

}

File: app/src/main/java/com/innso/mobile/ui/bindings/GeneralBindings.kt
package com.innso.mobile.ui.bindings

import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.innso.mobile.InnsoApplication
import com.innso.mobile.R
import java.io.File

object GeneralBindings {

    @JvmStatic
    @BindingAdapter("load_image", "place_holder")
    fun loadImage(imageView: ImageView, imagePath: String, placeHolder: Int? = null) {
        val context = InnsoApplication.get().applicationContext
        val options: RequestOptions = RequestOptions()
                .placeholder(placeHolder ?: R.mipmap.ic_launcher)
                .error(placeHolder ?: R.mipmap.ic_launcher)

        Glide.with(context).asBitmap()
                .load(imagePath)
                .apply(options)
                .into(imageView)
    }

    @JvmStatic
    @BindingAdapter("load_image_file")
    fun loadImageFile(imageView: ImageView, fileImage: File?) {
        if (fileImage != null) {
            Glide.with(imageView.context).load(fileImage).into(imageView)
        }
    }

}
